docs/ossm/ossm2-vs-ossm3.md

```diff
@@ -0,0 +1,103 @@
+
+## New resource: `IstioCNI`
+
+The Istio CNI node agent is used to configure traffic redirection for pods in the mesh. It runs as a DaemonSet, on every node, with elevated privileges. The Istio CNI agent has a lifecycle that is independent of Istio control planes, and must be upgraded separately. 
+
+In OpenShift Service Mesh 2, service meshes were namespace scoped by default and thus each mesh included its own instance of Istio CNI. As all OpenShift Service Mesh 3 meshes are cluster-wide, there can only be one version of Istio CNI per cluster, regardless of how many service mesh control planes are on the cluster.
+
+For these reasons, the OpenShift Service Mesh 3 operator manages Istio CNI with a separate resource called `IstioCNI`. A single instance of this resource is shared by all Istio control planes (managed by `Istio` resources). The `IstioCNI` resource must be upgraded before individual control planes (`Istio` resources) are upgraded.
+
+## Scoping of the Mesh: Discovery Selectors and labels replace `ServiceMeshMemberRoll` and `ServiceMeshMember`
+
+OpenShift Service Mesh 2 used the two resources `ServiceMeshMemberRoll` and `ServiceMeshMember` to indicate which namespaces were to be included in the mesh. When a mesh was created, it would only be scoped to the namespaces listed in the `ServiceMeshMemberRoll` or containing a `ServiceMeshMember` instance. This made it simple to include multiple service meshes in a cluster with each mesh tightly scoped, referred to as a “multitenant” configuration. 
+
+In OpenShift Service Mesh 2.4, a “cluster-wide” mode was introduced to allow a mesh to be cluster-scoped, with the option to limit the mesh using an Istio feature called `discoverySelectors`, which limits the scope of the mesh to a set of namespaces defined with a [label selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). This is aligned with how community Istio worked, and allowed Istio to manage cluster-level resources.
+
+OpenShift Service Mesh 3 converges with Istio in making all meshes “cluster-wide”. This means that Istio control planes are all cluster-scoped resources and the resources `ServiceMeshMemberRoll` and `ServiceMeshMember` are no longer present, with meshes scoped to the entire cluster by default and then scoped down using `discoverySelectors`.
+
+To replace `ServiceMeshMemberRoll`, a `discoverySelector` may be configured to limit the scope of the mesh to one or more namespaces with labels that match the `discoverySelector`'s label selector. 
+
+To replace `ServiceMeshMember`, label(s) may be added to namespaces such that they trigger the label selector specified by `discoverySelector`.
```

Comment:
This makes it look as though SMMR and SMM did two different things. But the SMM is just a way to add a namespace to the SMMR. Then, the SMMR affects both discovery and injection.

So, instead of saying that users can replace the SMMR with discoverySelectors, and replace the SMM with labels, you should rewrite this to talk about discovery and injection.

Reply:
I thought the purposes of the SMM was to let a namespace/project join the mesh without having permissions to edit SMMR... in the same way, a user may not have permissions to edit the `Istio` resource, but may be able to edit namespaces labels to include a namespace in the mesh via an already defined discoverySelector? 

In practice... I suspect it's always the same admin though.

There is a separate section on sidecar injection below. I am trying to separate discovery from injection as concepts, though one take away is that I should reconsider the term "scope" for "discovery", with the thinking that "scope" = "discovery" + "injection".

Reply:
I decided to remove these two sentences, and to your suggestion, add one describing injection and directing them to that section. I also reordered so that injection is the next topic.